test/integration_test.py

```diff
@@ -147,30 +147,35 @@ def tearDown(self):
     def test(self):
         if config.deployment_stage != 'prod':
             try:
-                self._test_indexing()
+                log.info('Starting test using test name, %s ...', self.test_name)
+                self.test_notifications, self.expected_fqids = self._test_notifications(test_name=self.test_name,
+                                                                                        test_uuid=self.test_uuid,
+                                                                                        max_bundles=self.max_bundles)
+                self.num_bundles = len(self.expected_fqids)
+                self.queues.wait_for_queue_level(empty=True,
+                                                 num_expected_bundles=self.num_bundles,
+                                                 min_timeout=self.min_timeout)
+                self.azul_client._index(self.test_notifications)
+                # Index some bundles again to test that we handle duplicate additions.
+                # Note: random.choices() may pick the same element multiple times so
+                # some notifications will end up being sent three or more times.
+                self.azul_client._index(random.choices(self.test_notifications, k=len(self.test_notifications) // 2))
+                self._check_bundles_are_indexed(self.test_name, 'files')
+            except Exception as e:
+                log.error('Exception while indexing. Cleaning up ...', exc_info=e)
+                self._delete_bundles(self.test_notifications)
+                raise e
```

Comment:
```suggestion
                raise
```

In previous versions of Python, there used to be a difference between these two ways of reraising, with respect to how the stack traces looked. To confirm/deny, please write a little test script that emulates this and post the source code and script output here.

Reply:
```py
def pho(di):
    return di['var']
di = {'a': 1}
try:
    pho()
except Exception as e:
    print('Did something')
        raise e
```
```console
>>> try:
...     pho(di)
... except Exception as e:
...     print('Did something')
...     raise e
... 
Did something
Traceback (most recent call last):
  File "<input>", line 5, in <module>
  File "<input>", line 2, in <module>
  File "<input>", line 2, in pho
KeyError: 'var'

>>> try:
...     pho(di)
... except Exception as e:
...     print('Did something')
...     raise
... 
Did something
Traceback (most recent call last):
  File "<input>", line 2, in <module>
  File "<input>", line 2, in pho
KeyError: 'var'
```

Reply:
Please do this as a script, and not just a console session. The console may alter the experiment. So to be precise there should be two scripts, one with `raise` and one with `raise e`. And the output from each.